Dockerfile

```diff
@@ -24,12 +24,19 @@ RUN yum install -y \
   curl tar \
   && yum clean all
 
-# install Oracle JRE
+# configure java runtime
 ENV JAVA_HOME=/opt/java \
   JAVA_VERSION_MAJOR=8 \
   JAVA_VERSION_MINOR=102 \
   JAVA_VERSION_BUILD=14
 
+# configure nexus runtime
+ENV NEXUS_VERSION=3.1.0-04 \
+  NEXUS_HOME=/opt/sonatype/nexus \
+  NEXUS_DATA=/nexus-data \
```

Comment:
`NEXUS_DATA` gets substituted with the actual path at image build time, so the user can't override it at container run time (e.g. `-e NEXUS_DATA=/foo/bar`). It only makes the path to which they map a bit shorter (i.e. `-v ~/nexus-data:/nexus-data` vs. `-v ~/nexus-data:/opt/sonatype/sonatype-work/nexus3`). I'm ok with leaving as `/nexus-data` for compatibility with earlier versions. Not sure it makes the image any more flexible and introduces what maybe unnecessary configuration massaging.